core/arch/arm/kernel/boot.c

```diff
@@ -956,7 +956,9 @@ static void init_primary(unsigned long pageable_part)
 		 */
 		assert(va && va <= boot_cached_mem_end);
 		boot_cached_mem_end = va;
+	}
 
+	if (IS_ENABLED(CFG_DYN_CONFIG)) {
```

Comment:
Will `virt_page_alloc()` be used when pager is enabled?

Reply:
The corresponding function with pager is `tee_pager_alloc()`. The pager manages all available physical memory that doesn't have to be mapped permanently. So it's a bit different. But as I transform the code to be more dynamic it might make sense to hide that behind an abstract interface.